### ISSUE TITLE
`stat` can result in unexpected runtime failures

### ISSUE THREAD

**Issue by Thimoteus** [purescript-node/purescript-node-fs#27]
[This](https://gist.github.com/Thimoteus/af1b459568100a0f3f33) is a failing example that's close to what I was using.

I'm not sure if the problem is with `stat` or `catchException`, but essentially this is the problem:

You can use `catchException` to discharge an `EXCEPTION` assumption from your program. Great! If we use enough `catchException`s, the compiler tells us (assuming everything was well-typed) that our program won't have any uncaught exceptions at runtime.

Except it does. `catchException` doesn't ... catch the exception ... when using `stat` on a path that doesn't exist.

This threw me for a loop since I was assuming that uncaught runtime exceptions wouldn't happen.

**Comment by hdgarrood** [comment]
:scream: this definitely shouldn't happen.

Perhaps we incorrectly assumed the error would be thrown immediately if `fs.statSync` failed? I'm not at my laptop right now but does it work in node using JS directly if you try to wrap `fs.statSync` in a try...catch block?

**Comment by Thimoteus** [comment]
It seems to work as intended in node:

```javascript
> try {
... fs.statSync('./something');
... } catch (e) {
... console.log(e);
... }
{ [Error: ENOENT: no such file or directory, stat './something']
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path: './something' }
```

**Comment by hdgarrood** [comment]
Fixed by https://github.com/purescript-node/purescript-node-fs/commit/b1a99f855a245e1031a91e33ff218b5531c0033e, released in 0.9.3. Thanks for the report! :)

**Comment by hdgarrood** [comment]
Oops, I meant v0.10.1. I really need to get on and implement `pulp version` to protect me from myself.
